src/main/java/org/reactome/release/qa/graph/PhysicalEntitiesCompartmentCheck.java

```diff
@@ -35,42 +37,59 @@ public String getDisplayName() {
     public QAReport executeQACheck() throws Exception {
         QAReport report = new QAReport();
 
-        String e2c = QACheckerHelper.getAttributeTableName(ReactomeJavaConstants.PhysicalEntity,
-                                                           ReactomeJavaConstants.compartment,
-                                                           dba);
-        String sql = "SELECT DB_ID" +
-                " FROM " + e2c +
-                " GROUP BY DB_ID" +
-                " HAVING COUNT(" + ReactomeJavaConstants.compartment + ") > 1";
-        Connection conn = dba.getConnection();
-        PreparedStatement ps = conn.prepareStatement(sql);
-        ResultSet rs = ps.executeQuery();
-        List<Long> dbIds = new ArrayList<Long>();
-        while (rs.next()) {
-            dbIds.add(rs.getLong(1));
-        }
-        Collection<GKInstance> entities =
-                dba.fetchInstances(ReactomeJavaConstants.PhysicalEntity, dbIds);
-        dba.loadInstanceAttributeValues(entities, LOAD_ATTS);
-        for (GKInstance entity: entities) {
-            if (isEscaped(entity)) {
-                continue;
-            }
-            // Only report complexes.
-            if (!entity.getSchemClass().isa(ReactomeJavaConstants.Complex)) {
-                continue;
-            }
-            GKInstance entityOnOtherCell = null;
-            if (entity.getSchemClass().isValidAttribute(JavaConstants.entityOnOtherCell))
-                entityOnOtherCell = (GKInstance) entity.getAttributeValue(JavaConstants.entityOnOtherCell);
-            if (entityOnOtherCell == null) {
-                report.addLine(entity.getDBID().toString(), 
-                               entity.getDisplayName(), 
-                               entity.getSchemClass().getName(), 
-                               QACheckerHelper.getLastModificationAuthor(entity));
+
+        for (SchemaClass sc : ((Collection<SchemaClass>) this.dba.getSchema().getClasses()))
```

Comment:
This may cause duplicated check. For example, EntitySet and DefinedSet will be both checked. Worse, there is no DefinedSet_2_compartment table. There is a method inside GKSchemaClass called getSubclasses(), which should be used to do a top-down check.